tools/hardklor/hardklor.xml

```diff
@@ -0,0 +1,75 @@
+    -maxcharge=$chMax -mslvl=$mslvl -correlation=$corr -depth=$depth -sensitivity=$sens -maxfeat=$maxfeat</command>
+    <inputs>
+	<param name="ms1_in" type="data" format="mzml" label="MS1 files with precursor spectra " />
+        <param name="res" type="integer" value="60000" label="Spectral resolution at 400m/z" />
+        <param name="centroided" type="select" display="radio" label="Centroided data?">
+          <option value="0" selected="true">No</option>
+          <option value="1">Yes</option>
+        </param>
+        <param name="instrument" type="select" label="Instrument" >
+            <option value="Orbitrap" selected="true">Orbitrap</option>
+            <option value="FTICR" >FTICR</option>
+            <option value="TOF" >TOF</option>
+            <option value="QIT" >Quadrupole ion trap</option>
+        </param>
+        <param name="mslvl" type="select" display="radio" label="MS level">
+          <option value="0" >All</option>
+          <option value="1" selected="true">MS1</option>
+          <option value="2">MS2</option>
+          <option value="3">MS3</option>
+        </param>
+        <param name="algorithm" type="select" label="Algorithm">
+            <option value="Basic">Basic</option>
+            <option value="Version1">Version 1</option>
+            <option value="Version2" selected="true">Version 2</option>
+        </param>
+        <param name="cdm" type="select" label="Charge determination method">
+            <option value="None">None</option>
+            <option value="FFT">Fast Fourier Transform</option>
+            <option value="Patterson">Patterson Algorithm</option>
+            <option value="Quick" selected="true">Quick</option>
+            <option value="Senko">Senko</option>
+        </param>
+        
+        <param name="chMin" type="integer" value="1" label="Min charge state" />
+        <param name="chMax" type="integer" value="5" label="Max charge state" />
+
+        <param name="corr" type="float" value="0.95" label="Correlation threshold" />
+        <param name="sens" type="integer" value="2" label="Sensitivity level" help="Ranges from 0 (min, fast) to 3 (max, slow)" />
+        <param name="depth" type="integer" value="2" label="Depth of combinatorial analysis" 
+help="Max number of overlapping features in any mz window. Keep as low as necessary for speed" />
+        <param name="maxfeat" type="integer" value="12" label="Max nr. of features for mz-window to solve" />
+
+<!--        <param name="sn" type="float" value="1" label="Signal to noise threshold" help="Must be >=0"  />
+        <param name="snWin" type="float" optional="true" label="Dynamic signal to noise window length (m/z)" 
+                help="For spectra with varying noise levels (not for Thermo data)"/>
+        <param name="win" type="float" value="5.25" label="Max width of any set of peaks in a spectrum" /> -->
+    </inputs>
+    
+    <outputs>
+        <data format="hk" name="output" />
```

Comment:
@glormph @bgruening   I tend to agree with Jorrit with subclassing tabular in this case.   There will be fewer help requests when users are guided to make sensible input choices rather than getting the long list of anything tabular.